Entities.js

```diff
@@ -0,0 +1,339 @@
+    input[3] = 1.5
+  }
+  const UndoLeft = () => {
+    input[0] = 0
+  }
+  const UndoRight = () => {
+    input[1] = 0  
+  }
+  const UndoUp = () => {
+    input[2] = 0
+  }
+  const UndoDown = () => {
+    input[3] = 0
+  }
+  UseOnKeyPress(Left, 'a')
+  UseOnKeyPress(Right,'d')
+  UseOnKeyPress(Up, 'w')
+  UseOnKeyPress(Down, 's')
+  UseOnKeyRelease(UndoLeft,'a')
+  UseOnKeyRelease(UndoRight,'d')
+  UseOnKeyRelease(UndoUp, 'w')
+  UseOnKeyRelease(UndoDown, 's')
+}
+
+//main player character
+class Knight{
+
+  constructor(){
+    this.height = playerHeight;
+    this.width = playerWidth;
+    this.backgroundColor = "blue";
+    this.x = 400;
+    this.y = 400;
+    this.velocityX = 0;
+    this.velocityY = 0;
+    this.Image = "./assets/Background.jpeg";
+    this.renderer = <CreateBrick/>;
+    this.momentum = [0,0];
+}
+// if x and y are presse
+//update knights position
+update(){
+  let horizontal = input[1] - input[0]
+  let vertical = input[3] - input[2]
+  if(Math.abs(this.momentum[0] + horizontal/4) > 4){
+    this.momentum[0] = Math.sign(this.momentum[0]) * 4
+  }
+  else{
+    this.momentum[0] += horizontal/4
+  }
+  if(Math.abs(this.momentum[1] + vertical/4) > 4){
+    this.momentum[1] = Math.sign(this.momentum[1]) * 4
+  }
+  else{
+    this.momentum[1] += vertical/4
+  }
+  //slow the knights velocity by a small amount (to lessen drifting)
+  this.momentum[0] -= .01 * Math.sign(this.momentum[0])
+  this.momentum[1] -= .01 * Math.sign(this.momentum[1])
+
+  this.x += this.momentum[0]
+  this.y += this.momentum[1]
+  //update the knights position and record the change
+  GlobalX = this.x
+  GlobalY = this.y
+}
+}
+
+
+class Brick {
+  constructor(id) {
+    this.id = id;
+    let x = Math.random()*500;
+    let y = 5;
+    this.x = x;
+    this.y = y;
+    this.backgroundColor = 'red';
+    this.height = 75;
+    this.width = 75;
+    this.Image = "./assets/Brick.jpeg";
+    this.renderer = <CreateBrick/>;
+    let randomNum = Math.random()*3;
+    this.velocityX = randomNum;
+    this.velocityY = 3 - randomNum;
+  }
+
+  update() {
+    //Bouncing off walls:
+    if(this.x >= (MAXwidth - this.width)){
+      this.reverse(-1,1)
+    }
+    if(this.x <= 0){
+      this.reverse(-1,1)
+    }
+    if(this.y >= (MAXheight - this.height)){
+      this.reverse(1,-1)
+    }
+    if(this.y <= 0){
+      this.reverse(1,-1)
+    }
+
+  //check for player intersections  
+    if(this.y + this.width < GlobalY  || this.y > GlobalY + playerHeight){
+    }
+    else if(this.x > GlobalX + playerWidth || this.x + this.width < GlobalX ){
+
+    }
+    else{
+      DeleteEntity(this.id)
+      counter += 1
+    }
+
+//updating the bricks position
+    this.x += this.velocityX;
+    this.y += this.velocityY;
+  }
+// TODO learn to use matterjs to handle collisions and possibly movement
+  reverse(xMult, yMult) {
+    this.velocityX = (this.velocityX * xMult)
+    this.velocityY = (this.velocityY * yMult)
+  }
+}
+
+class Circle {
```

Comment:
There is a lot of overlap and common definitions between Brick and Circle and Grenade (and maybe even Knight).
When you're using Classes, and they have a lot of similar behaviors, one thing you can use is inheritance:
```
// Define some common defaults for all shapes
// You can overwrite these definitions later if you need to
class Shape {
  constructor(x, y) {
    this.x = x;
    this.y = y;
    this.height = 75;
    this.width = 75;
    this.color = "red"
    this.momentum = [0, 0]
  }
  update() {
    // all shapes move and have momentum
    this.x += this.momentum[0];
    this.y += this.momentum[0];
  }
  sayHello() {
    console.log("Hello!")
  }
}

// Now we define a more specific shape, but inherit the definitions above
class Circle extends Shape {
  constructor(x, y) {
    // do common setup:
    super(x, y)
    // do specific updates
    this.renderer = <CreateCirlce />
    // you can just overwrite variables that are defined in the super's constructor:
    this.color = "blue"
  }
  update() {
     // do common update tasks:
     super.update()

     // do some circle specific updates:
     this.color = "green"
  }
}

let circ = new Circle(10, 10);
circ.x // value: 10 (set in super constructor)
circ.update() // first moves, then changes color
circ.sayHello() // prints "Hello!", (uses the "default" definition from the Shape class, even though we didn't define it for Circle)
```